In [3]:
import numpy as np
import pandas as pd
import matplotlib as plt
import seaborn as sns

# Einlesen der Daten
Auffällig sind schon die fehlenden Werte (Nan)

In [47]:
df = pd.read_csv("../data/immo_data_202208_v2.csv", low_memory=False)
df.drop(["Unnamed: 0.1","Unnamed: 0"], axis=1,inplace=True) #Zwei Spalten für Index ?
df.head()

Municipality Living space Plot area Floor space Availability  \
0   Biberstein       100 m²       NaN         NaN   On request   
1   Biberstein       156 m²    222 m²      242 m²   On request   
2          NaN          NaN       NaN         NaN          NaN   
3   Biberstein       154 m²    370 m²      257 m²   On request   
4     Küttigen       142 m²       NaN         NaN   On request   

                                location  \
0                    5023 Biberstein, AG   
1   Buhldenstrasse 8d5023 Biberstein, AG   
2                       5022 Rombach, AG   
3  Buhaldenstrasse 8A5023 Biberstein, AG   
4                       5022 Rombach, AG   

                                         description  \
0  3.5 rooms, 100 m²«Luxuriöse Attika-Wohnung mit...   
1  4.5 rooms, 156 m²«Stilvolle Liegenschaft - ruh...   
2  2.5 rooms, 93 m²«Moderne, lichtdurchflutete At...   
3  4.5 rooms, 154 m²«AgentSelly - Luxuriöses Eckh...   
4  4.5 rooms, 142 m²«MIT GARTENSITZPLATZ UND VIEL...   

                                detailed_description  \
0  DescriptionLuxuriöse Attika-Wohnung direkt an ...   
1  DescriptionStilvolle Liegenschaft an ruhiger L...   
2  detail_responsive#description_title2,5 Zimmerw...   
3  DescriptionDieses äusserst grosszügige Minergi...   
4  DescriptionAus ehemals zwei Wohnungen wurde ei...   

                                                 url  \
0  https://www.immoscout24.ch//en/d/penthouse-buy...   
1  https://www.immoscout24.ch//en/d/terrace-house...   
2  https://www.immoscout24.ch//en/d/penthouse-buy...   
3  https://www.immoscout24.ch//en/d/detached-hous...   
4  https://www.immoscout24.ch//en/d/flat-buy-romb...   

                                               table  ... features  \
0  b <article class=####Box-cYFBPY hKrxoH####><h2...  ...      NaN   
1  b <article class=####Box-cYFBPY hKrxoH####><h2...  ...      NaN   
2  b <article class=####Box-cYFBPY hKrxoH####><h2...  ...      NaN   
3  b <article class=####Box-cYFBPY hKrxoH####><h2...  ...      NaN   
4  b <article class=####Box-cYFBPY hKrxoH####><h2...  ...      NaN   

  description_detailed Floor space: Number of floors: Volume:     plz  \
0                  NaN          NaN               NaN     NaN  5023.0   
1                  NaN          NaN               NaN     NaN  5023.0   
2                  NaN          NaN               NaN     NaN  5022.0   
3                  NaN          NaN               NaN     NaN  5023.0   
4                  NaN          NaN               NaN     NaN  5022.0   

  Number of toilets: Gross yield: Minimum floor space: space_cleaned  
0                NaN          NaN                  NaN           NaN  
1                NaN          NaN                  NaN           NaN  
2                NaN          NaN                  NaN           NaN  
3                NaN          NaN                  NaN           NaN  
4                NaN          NaN                  NaN           NaN  

[5 rows x 132 columns]

In [48]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22481 entries, 0 to 22480
Columns: 132 entries, Municipality to space_cleaned
dtypes: float64(63), int64(2), object(67)
memory usage: 22.6+ MB


## Duplikate zählen

In [49]:
print(df.duplicated().sum())

0


Datensatz scheint keine Duplikate zu haben.

## Fehlende Daten zählen

### Insgesamt

In [57]:
print(df.isna().sum().sum())

1179856


Spalten mit den meisten fehlenden Werte

In [84]:
df.isna().sum().sort_values(ascending=False).head(10)

Minimum floor space:      22479
Superficie utile          22475
Gross return              22475
Piano                     22471
Superficie del terreno    22470
Surface utile             22467
Gross yield:              22467
Étage                     22457
Superficie abitabile      22454
Comune                    22454
dtype: int64

Spalten die komplett sind 

In [74]:
print((df.isna().sum() == 0).sum())

48


Im Vergleich zum ganzen Dataset

In [83]:
df.shape

(22481, 132)

**Zusammengefasst:** <br>
Es gibt sehr viele fehlende Werte. Wenn man die Spalten untersucht sieht man, dass es Saplten gibt wie "Minimum floor space:" welche 22479 fehlende Werte von 22481 hat. Es gibt aber auch 48 von 132 Spalten die keine fehlende Werte enthalten.  


In [64]:
df.loc[df["Minimum floor space:"].isna() == False]

Municipality Living space Plot area Floor space Availability  \
13553     Ecublens          NaN       NaN         NaN          NaN   
13554     Ecublens          NaN       NaN         NaN          NaN   

                                  location  \
13553  Chem. de Montaney 40, 1024 Ecublens   
13554  Chem. de Montaney 40, 1024 Ecublens   

                                             description  \
13553  Très bel appartement de 3 1/2 pièces sis dans ...   
13554  Très bel appartement de 3 1/2 pièces sis dans ...   

                                    detailed_description  \
13553  Description\n"Appartement en attique de 3 1/2 ...   
13554  Description\n"Appartement en attique de 3 1/2 ...   

                                          url table  ...  \
13553  https://www.homegate.ch/buy/3001883362   NaN  ...   
13554  https://www.homegate.ch/buy/3002001035   NaN  ...   

                                                features  \
13553  Pets allowed\tBalcony / Terrace\tWashing machi...   
13554  Pets allowed\tBalcony / Terrace\tWashing machi...   

                                    description_detailed Floor space:  \
13553  Description\n"Appartement en attique de 3 1/2 ...       100 m2   
13554  Description\n"Appartement en attique de 3 1/2 ...       100 m2   

      Number of floors: Volume:     plz Number of toilets: Gross yield:  \
13553               3.0     NaN  1024.0                NaN          NaN   
13554               3.0     NaN  1024.0                NaN          NaN   

      Minimum floor space: space_cleaned  
13553                88 m2         100.0  
13554                88 m2         100.0  

[2 rows x 132 columns]